collectors/ciscomeraki/collector.js

```diff
@@ -0,0 +1,270 @@
+ * @copyright (C) 2024, Alert Logic, Inc
+ * @doc
+ *
+ * ciscomeraki class.
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+'use strict';
+
+const moment = require('moment');
+const PawsCollector = require('@alertlogic/paws-collector').PawsCollector;
+const parse = require('@alertlogic/al-collector-js').Parse;
+const packageJson = require('./package.json');
+const calcNextCollectionInterval = require('@alertlogic/paws-collector').calcNextCollectionInterval;
+const utils = require("./utils");
+const AlLogger = require('@alertlogic/al-aws-collector-js').Logger;
+const MAX_POLL_INTERVAL = 900;
+let typeIdPaths = [{ path: ["type"] }];
+let tsPaths = [{ path: ["occurredAt"] }];
+
+class CiscomerakiCollector extends PawsCollector {
+    constructor(context, creds) {
+        super(context, creds, packageJson.version);
+    }
+
+    async pawsInitCollectionState(event, callback) {
+        const startTs = process.env.paws_collection_start_ts ?
+            process.env.paws_collection_start_ts :
+            moment().toISOString();
+        const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+
+        const resourceNames = process.env.collector_streams ? JSON.parse(process.env.collector_streams) : [];
+        try {
+            if (resourceNames.length > 0) {
+                const initialStates = resourceNames.map(networkId => ({
+                    networkId: networkId,
+                    since: startTs,
+                    until: endTs,
+                    nextPage: null,
+                    poll_interval_sec: 1
+                }));
+                return callback(null, initialStates, 1);
+            }
+            else {
+                let networks = await this.getAllNetworks(callback);
+                if (networks.length > 0) {
+                    const initialStates = networks.map(networkId => ({
+                        networkId: networkId,
+                        since: startTs,
+                        until: endTs,
+                        nextPage: null,
+                        poll_interval_sec: parseInt(Math.floor(Math.random() * 30) + 1)
+                    }));
+                    return callback(null, initialStates, 1);
+                } else {
+                    return callback("Error: No networks found");
+                }
+            }
+
+        } catch (error) {
+            return callback(error);
+        }
+    }
+
+    async getAllNetworks(callback) {
+        const { clientSecret, apiEndpoint, orgKey } = this.validateAndPrepare(callback);
+        const url = `/api/v1/organizations/${orgKey}/networks`;
+        try {
+            const networks = await utils.getAllNetworks(url, clientSecret, apiEndpoint);
+            return networks.map(network => network.id);
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async handleEvent(event) {
```

Comment:
I think this code may be we need to move to paws lib to handle new stream get added once collector get deployed.
But for now i think this should be part of index.js from where it called. What you think?

Reply:
not moved as expected